packages/tss-client/src/client.ts

```diff
@@ -185,78 +195,49 @@ export class Client {
-          })
-        );
-        this._readyResolves.push(clientResolve);
-        this._readyRejects.push(clientReject);
-        return;
-      }
-      if (socket.hasListeners("send")) {
-        socket.off("send");
-      }
-
-      // create pending promises for each server that resolves when precompute for that server is complete
-      let resolve: (value: unknown) => void;
-      let reject: (value: unknown) => void;
-      this._readyPromises.push(
-        // eslint-disable-next-line promise/param-names
-        new Promise((r, rj) => {
-          resolve = r;
-          reject = rj;
-        })
-      );
-      this._readyResolves.push(resolve);
-      this._readyRejects.push(reject);
-
-      // Add listener for incoming messages
-      socket.on("send", async (data, cb) => {
-        const { session, sender, recipient, msg_type, msg_data } = data;
-        if (session !== this.session) {
-          this.log(`ignoring message for a different session... client session: ${this.session}, message session: ${session}`);
-          return;
+      if (socket) {
+        if (socket.hasListeners("send")) {
+          socket.off("send");
         }
-        const pendingRead = this.pendingReads.get(`session-${session}:sender-${sender}:recipient-${recipient}:msg_type-${msg_type}`);
-        if (pendingRead !== undefined) {
-          // globalThis.total_incoming += msg_data.length;
-          // globalThis.total_incoming_msg.push(msg_data);
-          pendingRead(msg_data);
-        } else {
+
+        // Add listener for incoming messages
+        socket.on("send", async (data, cb) => {
+          const { session, sender, recipient, msg_type, msg_data } = data;
+          if (session !== this.session) {
+            this.log(`ignoring message for a different session... client session: ${this.session}, message session: ${session}`);
+            return;
+          }
           this.msgQueue.push({ session, sender, recipient, msg_type, msg_data });
-        }
-        if (cb) cb();
-      });
-      // Add listener for completion
-      socket.on("precompute_complete", async (data, cb) => {
-        const { session, party } = data;
-        if (session !== this.session) {
-          this.log(`ignoring message for a different session... client session: ${this.session}, message session: ${session}`);
-          return;
-        }
-        if (cb) cb();
-        this.precomputes[this.parties.indexOf(party)] = "precompute_complete";
-        resolve(null);
-      });
+          if (cb) cb();
+        });
+        // Add listener for completion
+        socket.on("precompute_complete", async (data, cb) => {
+          const { session, party } = data;
+          if (session !== this.session) {
+            this.log(`ignoring message for a different session... client session: ${this.session}, message session: ${session}`);
+            return;
+          }
+          this._precomputeComplete.push(party);
+          if (cb) cb();
```

Comment:
any reason why cb is called after updating _precomputeComplete array in success case but its opposite in failure?

Reply:
Thanks, they should be identical, have fixed it.